PoC/kafka-pull/kafka-broker-extension/src/main/java/org/eclipse/tractusx/edc/extensions/kafka/KafkaBrokerDataFlowController.java

```diff
@@ -20,131 +21,176 @@
+        var tokenUrl = contentDataAddress.getStringProperty(OAUTH_TOKEN_URL);
+        var revokeUrl = Optional.ofNullable(contentDataAddress.getStringProperty(OAUTH_REVOKE_URL));
+        var clientId = contentDataAddress.getStringProperty(OAUTH_CLIENT_ID);
+        var clientSecret = getSecret(contentDataAddress.getStringProperty(OAUTH_CLIENT_SECRET_KEY));
+
+        return new OAuthCredentials(tokenUrl, revokeUrl, clientId, clientSecret);
     }
 
     @Override
-    public StatusResult<Void> suspend(TransferProcess transferProcess) {
+    public StatusResult<Void> suspend(final TransferProcess transferProcess) {
         return deleteCredentialsAndRevokeAccess(transferProcess, SUSPEND_FAILED);
     }
 
     @Override
-    public StatusResult<Void> terminate(TransferProcess transferProcess) {
+    public StatusResult<Void> terminate(final TransferProcess transferProcess) {
         return deleteCredentialsAndRevokeAccess(transferProcess, TERMINATE_FAILED);
     }
 
     @Override
-    public Set<String> transferTypesFor(Asset asset) {
+    public Set<String> transferTypesFor(final Asset asset) {
         return Set.of(TRANSFER_TYPE);
     }
 
-    private String getSecret(String secretKey) {
+    private String getSecret(final String secretKey) {
         return Optional.ofNullable(vault.resolveSecret(secretKey)).orElseThrow(() -> new EdcException(SECRET_NOT_DEFINED.formatted(secretKey)));
     }
 
-    private StatusResult<Void> deleteCredentialsAndRevokeAccess(TransferProcess transferProcess, String error) {
+    private StatusResult<Void> deleteCredentialsAndRevokeAccess(final TransferProcess transferProcess, final String error) {
         try {
             var contentDataAddress = transferProcess.getContentDataAddress();
-            var topic = contentDataAddress.getStringProperty(TOPIC);
-            var username = transferProcess.getId();
-            var secret = getSecret(contentDataAddress.getStringProperty(SECRET_KEY));
-
-            try (var kafkaAdminService = kafkaAdminServiceProvider.provide(contentDataAddress, secret)) {
-                kafkaAdminService.deleteCredentialsAndRevokeAccess(username, topic);
-                vault.deleteSecret(username);
-                return StatusResult.success();
-            }
-        } catch (ExecutionException | InterruptedException | EdcException | TimeoutException e) {
-            return StatusResult.failure(ResponseStatus.FATAL_ERROR, error + e.getMessage());
+            var transferProcessId = transferProcess.getId();
+
+            OAuthCredentials creds = extractOAuthCredentials(contentDataAddress);
+            String token = getSecret(transferProcessId);
+
+            oauthService.revokeToken(creds, token);
+            vault.deleteSecret(transferProcessId);
+
+            return StatusResult.success();
+
+        } catch (EdcException e) {
+            return StatusResult.failure(FATAL_ERROR, error + e.getMessage());
```

Comment:
Think about formatting. Seperator?

Reply:
The defined message of error already contains the relevant formatting